samples/matter/lock/src/combine_ota_images.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+"""
+Utility for combining Thread and Wi-Fi application binary images into one Matter OTA compliant file
+"""
+
+import sys
+import argparse
+import tempfile
+import os
+import cbor2
+import struct
+
+NCS_PATH = os.path.join(os.environ.get('ZEPHYR_BASE'), '../')
+sys.path.append(os.path.join(NCS_PATH, 'modules/lib/matter/src/app'))
+sys.path.append(os.path.join(NCS_PATH, 'nrf/scripts/bootloader'))
+
+import ota_image_tool as ota
+import dfu_multi_image_tool as multi
+
+# ota_image_tool args translators
+class ParseHeaderArgs:
+    def __init__(self, imageFile,):
+        self.image_file = imageFile
+
+class RemoveHeaderArgs:
+    def __init__(self, imageFile, outputFile):
+        self.image_file = imageFile
+        self.output_file = outputFile
+
+tlvHeaderKeysMap = {
+    ota.HeaderTag.VENDOR_ID : 'vendor_id',
+    ota.HeaderTag.PRODUCT_ID : 'product_id',
+    ota.HeaderTag.VERSION : 'version',
+    ota.HeaderTag.VERSION_STRING : 'version_str',
+    ota.HeaderTag.MIN_VERSION : 'min_version',
+    ota.HeaderTag.MAX_VERSION : 'max_version',
+    ota.HeaderTag.RELEASE_NOTES_URL : 'release_notes',
+    ota.HeaderTag.DIGEST_TYPE : 'digest_algorithm'
+}
+
+class GenerateOtaImageArgs:
+    def __init__(self, tlv: dict, input_file: str, output_file: str):
+        for key in tlvHeaderKeysMap:
+            valueToSet = tlv[key] if key in tlv else None
+            if tlvHeaderKeysMap[key] == 'digest_algorithm':
+                valueToSet = next(key for key, value in ota.DIGEST_ALGORITHM_ID.items() if value == valueToSet) # inverse lookup
+            self.__dict__[tlvHeaderKeysMap[key]] = valueToSet
+        self.input_files = [input_file]
+        self.output_file = output_file
+
+
+# Helper functions
+def remove_ota_header(imageFilePath: str) -> bytes:
+    """
+    Return binary representation of OTA DFU image file with Matter specific OTA header removed
+    """
+    with tempfile.TemporaryDirectory() as tempDir:
+        outputFile = os.path.join(tempDir, 'outputFile')
+        args = RemoveHeaderArgs(imageFilePath, outputFile)
+        ota.remove_header(args)
+        with open(args.output_file, 'rb') as extractedImageFile:
+            return extractedImageFile.read()
+
+def parse_multi_image_header(image: bytes) -> object:
+    """
+    Return JSON representation of NRF multi image DFU header extracted from input bytes object
+    """
+    with tempfile.TemporaryDirectory() as tempDir:
```

Comment:
Would `return multi.parse_header(io.BytesIO(tempImgFile))` work as well? Similar can be done for parsing OTA image file.

Reply:
Ahh yes, this actually something I wanted to try but then forgot about it. I event thought that the whole interaction between the `combine_ota_images.py` and existing scripts can be implemented with the io.BytesIO API, but unfortunately the latter tend to open a file under a path provided with args - looks like they were not written with reusability in mind.

Reply:
When it comes to `dfu_multi_image_tool.py`, you're free to modify it so that some function e.g. for parsing the header take a file object rather than path. `ota_image_tool.py` can be refactor in a similar manner in the future.

Reply:
Ok, for now I will file a low priority Jira issue for that as this task already takes too long. I guess after modification of `dfu_multi_image_tool.py` we would need to perform some thorough testing not to affect other dfu_multi_image library users.

Reply:
https://projecttools.nordicsemi.no/jira/browse/KRKNWK-16132